libraries/stdlib/jvm/src/kotlin/text/CharJVM.kt

```diff
@@ -107,13 +171,55 @@ public actual inline fun Char.toLowerCase(): Char = Character.toLowerCase(this)
 public inline fun Char.isTitleCase(): Boolean = Character.isTitleCase(this)
 
 /**
- * Converts this character to titlecase.
+ * Converts this character to title case using Unicode mapping rules of the invariant locale.
  *
  * @see Character.toTitleCase
- * @sample samples.text.Chars.toTitleCase
  */
+@OptIn(ExperimentalStdlibApi::class)
+@kotlin.internal.InlineOnly
+public inline fun Char.toTitleCase(): Char = titlecaseChar()
+
+/**
+ * Converts this character to title case using Unicode mapping rules of the invariant locale.
+ *
+ * This function performs one-to-one character mapping using case mapping information from the UnicodeData.txt file,
+ * to support one-to-many character mapping use the [titlecase] function.
+ * If this character has no mapping equivalent, the result of calling [uppercaseChar] is returned.
+ *
+ * @sample samples.text.Chars.titlecase
+ */
+@SinceKotlin("1.4")
+@ExperimentalStdlibApi
+@kotlin.internal.InlineOnly
+public inline fun Char.titlecaseChar(): Char = Character.toTitleCase(this)
+
+/**
+ * Converts this character to title case using Unicode mapping rules of the invariant locale.
+ *
+ * If this character has no mapping equivalent, the result of calling [uppercase] is returned.
+ *
+ * @sample samples.text.Chars.titlecase
+ */
+@SinceKotlin("1.4")
+@ExperimentalStdlibApi
+@kotlin.internal.InlineOnly
+public fun Char.titlecase(): String {
+    return titlecaseChar().let { if (it != uppercaseChar()) it.toString() else uppercase() }
```

Comment:
I believe `Chat.titlecase` should be implemented as following:
- find `u = it.uppercase()`, then if it is multi-char, return `u.substring(0, 1) + u.substring(1).lowercase()`
- otherwise return `titlecaseChar.toString()`

Reply:
done